### ISSUE TITLE
Alias via using within nested namespace results in incorrect symbol name.

### ISSUE THREAD

**Issue by cantonios** [doxygen/doxygen#11281]
**Describe the bug**

```cpp
namespace Eigen {
namespace indexing {
using Eigen::placeholders::all;  // <--- Doxygen call this symbol Eigen::indexing::placeholders::all
}
}
```
When writing doxytags, the above symbol is recorded as:
```xml
<compound kind="namespace">
    <name>Eigen::indexing</name>
    <member kind="variable" static="yes">
        <name>placeholders::all</name>
    </member>
</compound>
```
Using the doxytags to link into the documentation then fails to find the symbol `Eigen::indexing::all`.

**To Reproduce**

[doxytags.tar.gz](https://github.com/user-attachments/files/18172417/doxytags.tar.gz)

**Expected behavior**

The symbol should not include `placeholders::`.

**Version**
Compiled from `master` commit 163ed0328

**Potential Fix**

I'm wondering if [this](https://github.com/doxygen/doxygen/blob/163ed03282c894bd9a15cc2a81182cb5336daefd/src/doxygen.cpp#L2224) should be
```cpp
- int i=root->name.find("::");
+ int i=root->name.findRev("::");
```
This corrects the symbol names and allows all doxygen's tests to continue passing.

---

Updated example: 
[doxytags.tar.gz](https://github.com/user-attachments/files/18214723/doxytags.tar.gz)

```
$ doxygen supported.cfg 1> supported.log
$ doxygen unsupported.cfg 1> unsupported.log
unsupported/doc/unsupported.md:3: warning: unable to resolve reference to 'Eigen::indexing::all' for \ref command
unsupported/doc/unsupported.md:3: warning: unable to resolve reference to 'Eigen::indexing::all' for \ref command
```

**Comment by albert-github** [comment]
I only see that you are generating a tags file but I don't see you using it, so I don't understand what you mean by:
> Using the doxytags to link into the documentation then fails to find the symbol `Eigen::indexing::all`.

- Where should the link be ?
- When I run the example with and without `GENERATE_TAGFILE` I get the same html output, so how is this related to the tag file?

**Comment by cantonios** [comment]
The symbol name itself stored within doxygen is incorrect.  The [`DefinitionImpl::Private::name`](https://github.com/doxygen/doxygen/blob/163ed03282c894bd9a15cc2a81182cb5336daefd/src/definition.cpp#L101) is `placeholders::all`, under the [`outerScope`](https://github.com/doxygen/doxygen/blob/163ed03282c894bd9a15cc2a81182cb5336daefd/src/definition.cpp#L91) of namespace definition `Eigen::indexing`.  This is wrong, and is the reason why the tags file is wrong.

> I only see that you are generating a tags file but I don't see you using it

Since it's incorrect in the doxytags file, any _other_ project that tries to link to it via the tags file (i.e. https://www.doxygen.nl/manual/external.html) will fail to find the symbol `Eigen::indexing::all`, because according to the doxytags file, that symbol doesn't exist.  In our case, we have separate "sites" for Eigen's [supported](https://libeigen.gitlab.io/docs/index.html) modules, and [unsupported](https://libeigen.gitlab.io/docs/unsupported/index.html) modules.  Links are broken in the unsupported site.  But the issue itself is the bad doxytags file.

**Comment by albert-github** [comment]
The problem might be an incorrect tags file but a complete example (generating the tags file an using the tags file) is necessary to check what can be done and can be tested (!). So:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

**Comment by cantonios** [comment]
Added another example above with two projects.  The second project doesn't tell us anything new though - it just outputs:
```
unsupported/doc/unsupported.md:3: warning: unable to resolve reference to 'Eigen::indexing::all' for \ref command
```
because the doxytags file generated from the first project is incorrect.

**Comment by albert-github** [comment]
Is the code valid c++ code?
I see:
```
namespace indexing {
/** Alias for placeholders. */
using Eigen::placeholders::all;
}
```
and this is already inside the `namespace Eigen`.

When I change this to:
```
namespace indexing {
/** Alias for placeholders. */
using placeholders::all;
}
```
I don't get the warning anymore.

**Comment by cantonios** [comment]
> Is the code valid c++ code? I see:
> 
> ```
> namespace indexing {
> /** Alias for placeholders. */
> using Eigen::placeholders::all;
> }
> ```

Yes, this is valid code.

> When I change this to:
> 
> ```
> namespace indexing {
> /** Alias for placeholders. */
> using placeholders::all;
> }
> ```
> 
> I don't get the warning anymore.

You don't get the warning because now the name has only  one `::` in it.  You would still have the issue if it were:
```cpp
namespace Eigen {
namespace internal {
namespace impl {
static all_t all;
}  // namespace impl
}  // namespace internal

namespace indexing {
using internal::impl::all;
}  // namespace indexing

}  // namespace Eigen
```

**Comment by doxygen** [comment]
@cantonios I think you are right about the change from `find` to `findRev`. Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by albert-github** [comment]
Looks like to work now, though the  `TAGFILES` line  in unsupported.cfg should tread:
```
TAGFILES               = supported/supported.doxytags=../../supported/html
```

**Comment by cantonios** [comment]
Yes, that fixes it, thank you!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
